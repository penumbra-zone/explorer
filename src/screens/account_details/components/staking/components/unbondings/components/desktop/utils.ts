export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 25,
  },
  {
    key: 'commission',
    align: 'right',
    width: 25,
  },
  {
    key: 'amount',
    width: 25,
    align: 'right',
  },
  {
    key: 'linkedUntil',
    width: 25,
    align: 'right',
  },
];
