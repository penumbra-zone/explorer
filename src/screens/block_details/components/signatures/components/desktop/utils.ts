export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'validator',
    width: 34,
  },
  {
    key: 'votingPower',
    align: 'right',
    width: 33,
  },
  {
    key: 'signed',
    width: 33,
  },
];
