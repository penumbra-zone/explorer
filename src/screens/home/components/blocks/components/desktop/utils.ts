export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
}[] = [
  {
    key: 'height',
  },
  {
    key: 'validator',
  },
  {
    key: 'hash',
  },
  {
    key: 'txs',
  },
  {
    key: 'time',
    align: 'right',
  },
];
