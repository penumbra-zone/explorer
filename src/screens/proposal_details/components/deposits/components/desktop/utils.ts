export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'depositor',
    width: 34,
  },
  {
    key: 'amount',
    width: 33,
    align: 'right',
  },
  {
    key: 'time',
    width: 33,
  },
];
