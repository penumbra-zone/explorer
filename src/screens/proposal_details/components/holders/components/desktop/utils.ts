export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 10,
  },
  {
    key: 'address',
    width: 30,
  },
  {
    key: 'quantity',
    width: 20,
    align: 'right',
  },
  {
    key: 'percentage',
    width: 20,
  },
  {
    key: 'value',
    align: 'right',
    width: 20,
  },
];
