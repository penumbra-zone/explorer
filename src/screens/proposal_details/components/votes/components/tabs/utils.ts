export const tabLabels = [
  {
    key: 'all',
    num: 30,
  },
  {
    key: 'yes',
    num: 3,
  },
  {
    key: 'no',
    num: 3,
  },
  {
    key: 'noWithVeto',
    num: 3,
  },
];
