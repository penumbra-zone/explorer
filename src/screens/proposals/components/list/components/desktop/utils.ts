export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'idx',
    width: 15,
  },
  {
    key: 'token',
    width: 10,
  },
  {
    key: 'price',
    align: 'right',
    width: 15,
  },
  {
    key: 'change',
    align: 'right',
    width: 15,
  },
  {
    key: 'volume',
    align: 'right',
    width: 15,
  },
  {
    key: 'marketCap',
    align: 'right',
    width: 15,
  },
  {
    key: 'holders',
    align: 'right',
    width: 15,
  },
];
