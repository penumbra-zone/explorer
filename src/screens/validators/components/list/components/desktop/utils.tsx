import React from 'react';
import {
  Typography,
} from '@material-ui/core';
import { InfoPopover } from '@components';

export const fetchColumns = (t): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sort?: boolean;
}[] => {
  return ([
    {
      key: 'idx',
      width: 10,
    },
    {
      key: 'validator',
      width: 20,
      sort: true,
    },
    {
      key: 'stake',
      align: 'right',
      width: 15,
      sort: true,
    },
    {
      key: 'fee',
      align: 'right',
      width: 10,
      sort: true,
    },
    {
      key: 'lastVote',
      align: 'right',
      width: 15,
    },
    {
      key: 'skipRate',
      width: 15,
      sort: true,
    },
    {
      key: 'condition',
      align: 'center',
      width: 15,
      component: (
        <Typography variant="h4" className="label popover">
          {t('condition')}
          <InfoPopover
            content={t('conditionExplanation')}
          />
        </Typography>
      ),
    },
  ]);
};
