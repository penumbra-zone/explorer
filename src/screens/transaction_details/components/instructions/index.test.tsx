import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Instructions from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
  Tag: (props) => <div id="Tag" {...props} />,
}));

jest.mock('react-virtualized-auto-sizer', () => ({ children }: any) => children({
  height: 600, width: 600,
}));

jest.mock('../../contexts/transaction', () => ({
  useTransactionContext: () => ({
    item: {
      slot: '812,768,640',
      signature: '4SGxuRMcseNbwki3tGxXPpfz7iFnuo9FUpTfiM4gJ8rhH59uZYSBBK2zW27xRdGX8Sb2N4VkGUnBYt59SBKEhPfB',
      amount: 123,
      success: true,
      time: 1615187146246,
      instructions: Array(2).fill({
        parsed: {
          info: {
            destination: 'AXUChvpRwUUPMJhA4d23WcoyAL7W8zgAeo7KoH57c75F',
            lamports: 8287,
            source: 'AXUChvpRwUUPMJhA4d23WcoyAL7W8zgAeo7KoH57c75F',
          },
          type: 'transfer',
        },
        program: 'system',
        programId: '11111111111111111111111111111111',
      }),
    },
  }),
}));

// ==================================
// unit tests
// ==================================
describe('screen: Home/Blocks/Mobile', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Instructions />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
