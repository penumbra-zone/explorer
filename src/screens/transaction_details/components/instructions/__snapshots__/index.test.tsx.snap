// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Blocks/Mobile matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Box"
>
  <h2
    className="MuiTypography-root MuiTypography-h2"
  >
    transactions:instructions
  </h2>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="makeStyles-list"
  >
    <div
      className="List"
      onScroll={[Function]}
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "direction": "ltr",
          "height": 600,
          "overflow": "auto",
          "position": "relative",
          "width": 600,
          "willChange": "transform",
        }
      }
    >
      <div
        style={
          Object {
            "height": 200,
            "pointerEvents": undefined,
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "height": 100,
              "left": 0,
              "position": "absolute",
              "right": undefined,
              "top": 0,
              "width": "100%",
            }
          }
        >
          <div>
            <div
              className="makeStyles-fakeItem"
            >
              <div
                className="makeStyles-tags"
              >
                <div
                  id="Tag"
                  theme="two"
                  value="system"
                />
                <div
                  id="Tag"
                  theme="two"
                  value="transfer"
                />
              </div>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent libero dolor, sollicitudin fringilla augue gravida, tincidunt viverra odio. Curabitur sit amet erat ultricies nisi posuere bibendum.
              </p>
            </div>
            <hr
              className="MuiDivider-root"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "height": 100,
              "left": 0,
              "position": "absolute",
              "right": undefined,
              "top": 100,
              "width": "100%",
            }
          }
        >
          <div>
            <div
              className="makeStyles-fakeItem"
            >
              <div
                className="makeStyles-tags"
              >
                <div
                  id="Tag"
                  theme="two"
                  value="system"
                />
                <div
                  id="Tag"
                  theme="two"
                  value="transfer"
                />
              </div>
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent libero dolor, sollicitudin fringilla augue gravida, tincidunt viverra odio. Curabitur sit amet erat ultricies nisi posuere bibendum.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
